### PR TITLE
Correctly identify identifiers inside JSX exprs as expression identifiers

### PR DESCRIPTION
**Pull request by RyanCavanaugh**
Fixes bug #3928.

I am inclined to start a campaign to make as many of our `switch` statements in the compiler exhaustive as possible?

### CODE REVIEW

**Comment by danquirk** [comment]
Exhaustiveness checking would be good :)

File: src/compiler/emitter.ts [hunk 1]
```diff
@@ -1484,6 +1485,7 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
                     }
                 }
                 else if (languageVersion < ScriptTarget.ES6) {
+                    debugger;
```
Comment (s-panferov) [inline src/compiler/emitter.ts]: Should be removed?

Comment (RyanCavanaugh) [inline src/compiler/emitter.ts]: Oops

File: tests/cases/compiler/jsxImportInAttribute.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+//@jsx: preserve
+//@module: commonjs
+
+//@filename: component.d.ts
+declare module "Test" {
+	export default class Text { }
```
Comment (DanielRosenwasser) [inline tests/cases/compiler/jsxImportInAttribute.tsx]: For the future, can you set your user settings to `"translate_tabs_to_spaces": true`?

**Comment by DanielRosenwasser** [comment]
:shipit:

**Pull request merged**